graftm/hmmer.py

```diff
@@ -545,144 +490,212 @@ def extract_orfs(self, input_path, orfm, orf_out_path):
+                      restrict_read_length=restrict_read_length)
         
-        if search_method == 'hmmsearch':
+        if search_method == 'hmmsearch': 
+            # run hmmsearch
             search_result = self.hmmsearch(
                                            hmmsearch_output_table,
                                            raw_reads,
                                            unpack,
-                                           args.input_sequence_type,
-                                           args.threads,
-                                           args.eval,
+                                           unpack.sequence_type(),
+                                           threads,
+                                           evalue,
                                            orfm
                                            )
-            for result in search_result:
-                hit_readnames=[x[0] for x in result.each([SequenceSearchResult.QUERY_ID_FIELD])]
-        
+
         elif search_method == 'diamond':
-            #run diamond
-            search_result =  Diamond(
+            # run diamond
+            search_result = Diamond(
                                      database=os.path.join(
-                                                           gpkg._base_directory, \
-                                                           gpkg._contents_hash[gpkg.DIAMOND_DATABASE_KEY]
+                                                           gpkg.base_directory, \
+                                                           gpkg.contents_hash[gpkg.DIAMOND_DATABASE_KEY]
                                                            ),
-                                     threads=args.threads,
-                                     evalue=args.eval,
+                                     threads=threads,
+                                     evalue=evalue,
                                      ).run(
                                            raw_reads,
-                                           args.input_sequence_type
+                                           unpack.sequence_type()
                                            )
-            search_result=[search_result]
-            for result in search_result:
-                hit_readnames=[x[0] for x in result.each([SequenceSearchResult.QUERY_ID_FIELD])]
-        else:
+            search_result = [search_result]
+
+        else:  # if the search_method isn't recognised
             raise Exception("Programming error: unexpected search_method %s" % search_method)
+        
         with tempfile.NamedTemporaryFile(prefix='graftm_readnames') as readnames:
-            orfm_regex = re.compile('^(\S+)_(\d+)_(\d)_(\d+)')
-
+            # Write the names of hits to a tmpfile
+            
+            orfm_regex = re.compile('^(\S+)_(\d+)_(\d)_(\d+)')  # to remove OrfM suffix from read names
+            if srch_aln_only:
+                hits = {key:[] for key in list(
```

Comment:
There's unecessary `list()`s here. But more to the point a regular for loop would be much easier to understand. e.g.
```
for result in search_result:
for array in result.each([SequenceSearchResult.QUERY_ID_FIELD]):
hits[array[0]] = []
```